From cb047813baa80505906c46749461f315aad0c901 Mon Sep 17 00:00:00 2001
From: Yurical <yurical1g@gmail.com>
Date: Fri, 16 Feb 2024 10:15:18 +0900
Subject: [PATCH] base: Remove storage redirect

---
 .../ExternalStorageProvider.java              | 32 -------------------
 1 file changed, 32 deletions(-)

diff --git a/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java b/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
index 3409c29d3c2c..18167a8a4a7f 100644
--- a/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
+++ b/packages/ExternalStorageProvider/src/com/android/externalstorage/ExternalStorageProvider.java
@@ -327,38 +327,6 @@ public class ExternalStorageProvider extends FileSystemProvider {
     @Override
     protected boolean shouldBlockDirectoryFromTree(@NonNull String documentId)
             throws FileNotFoundException {
-        final File dir = getFileForDocId(documentId, false);
-        // The file is null or it is not a directory
-        if (dir == null || !dir.isDirectory()) {
-            return false;
-        }
-
-        // Allow all directories on USB, including the root.
-        if (isOnRemovableUsbStorage(documentId)) {
-            return false;
-        }
-
-        // Get canonical(!) path. Note that this path will have neither leading nor training "/".
-        // This the root's path will be just an empty string.
-        final String path = getPathFromDocId(documentId);
-
-        // Block the root of the storage
-        if (path.isEmpty()) {
-            return true;
-        }
-
-        // Block /Download/ and /Android/ folders from the tree.
-        if (equalIgnoringCase(path, Environment.DIRECTORY_DOWNLOADS) ||
-                equalIgnoringCase(path, Environment.DIRECTORY_ANDROID)) {
-            return true;
-        }
-
-        // This shouldn't really make a difference, but just in case - let's block hidden
-        // directories as well.
-        if (shouldHideDocument(documentId)) {
-            return true;
-        }
-
         return false;
     }
 
-- 
2.43.0

